libmamba/src/filesystem/u8path.cpp

```diff
@@ -6,21 +6,37 @@
 
 #include <filesystem>
 #include <string>
+#if !defined(_WIN32)
+#include <fcntl.h>
+#include <sys/stat.h>
 
-#include "mamba/core/mamba_fs.hpp"
-#include "mamba/util/string.hpp"
+// We can use the presence of UTIME_OMIT to detect platforms that provide
+// utimensat.
+#if defined(UTIME_OMIT)
+#define USE_UTIMENSAT
+#endif
+#endif
+
+#include "mamba/filesystem/u8path.hpp"
```

Comment:
BTW the usual way to organize header is that in the cpp matching a header we put the related header as first. Currently our clang-format will reorganize them in alphabetical order. I was wondering: is there a way to have both logics?

Reply:
Yes, I think clang-format does that if the header is written as `"header.hpp"` (no directory)

Reply:
But then it seems to not always compile...

Reply:
😬 not a good sign then... will be fixed by modules one day 🥲